python/py_helpers.test.py

```diff
@@ -254,6 +279,47 @@ def foo():
 
         self.assertFalse(node.find_function("foo").has_args("int"))
 
+    def test_find_calls(self):
+        code_str = """
+print(1)
+int("1")
+print(2)
+obj.foo("spam")
+"""
+        node = Node(code_str)
+
+        self.assertEqual(len(node.find_calls("print")), 2)
+        self.assertTrue(node.find_calls("print")[0].is_equivalent("print(1)"))
+        self.assertTrue(node.find_calls("print")[1].is_equivalent("print(2)"))
+        self.assertEqual(len(node.find_calls("int")), 1)
+        self.assertTrue(node.find_calls("int")[0].is_equivalent("int('1')"))
+        self.assertEqual(len(node.find_calls("foo")), 1)
+        self.assertTrue(node.find_calls("foo")[0].is_equivalent("obj.foo('spam')"))
```

Comment:
It's a little weird that this works, but I think it's okay. Just to confirm: this only checks one level deep? i.e. it finds `obj.foo()`, but not `obj.foo.bar()`?

Reply:
> it finds obj.foo(), but not obj.foo.bar()?
It finds both.

It's a bit weird but I couldn't find a way to reach the same result without complicating things. Do you think `find_calls('foo')` should not find `obj.foo()` or `obj.foo.bar()`?

Reply:
That's okay, I don't think this approach is bad, but it might need a bit more documentation to explain what it does.

> Do you think find_calls('foo') should not find obj.foo() or obj.foo.bar()?

It's fine to find `obj.foo()`, I think, but not `obj.foo.bar()`. We can deal with `obj.bar.foo()` if/when you want to.

Reply:
I've just realized that I misread your comment. It finds `obj.foo()` as well as `obj.bar.foo()`, not `obj.foo.bar()`.

Do you still think it's a problem?

Reply:
My mistake. I didn't realise that it was already finding `obj.bar.foo()`.

No, I don't think that's a problem, but I do think it's worth mentioning in the docs. It's slightly surprising (to me, at least) that you don't have to traverse the object i.e. write something like `Node("obj.a.b").find_attr("a").find_calls("b")`.

Reply:
It was surprising to me as well. But the function call is always stored as the `attr` of `func`:

```
Module(
  body=[
    Expr(
      value=Call(
        func=Attribute(
          value=Attribute(
            value=Attribute(
              value=Attribute(
                value=Name(id='a', ctx=Load()),
                attr='b',
                ctx=Load()),
              attr='c',
              ctx=Load()),
            attr='d',
            ctx=Load()),
          attr='foo',
          ctx=Load()),
        args=[],
        keywords=[]))],
  type_ignores=[])
```

Reply:
Yeah, I did the same thing to check.

It makes perfect sense if you look at the AST, but our api isn't simply a wrapper over the AST. It has its own conventions and this is slight departure from those conventions. It's probably fine to do that, since the function you've designed is nice and easy to use, but it is worth mentioning what's going on.